python/ray/data/dataset.py

```diff
@@ -877,6 +877,16 @@ def split(
         if n <= 0:
             raise ValueError(f"The number of splits {n} is not positive.")
 
+        if equal:
+            count = self.count()
+            split_index = count // n
+            # we are creating n split_indices which will generate
+            # n + 1 splits; the last split will at most contains (n - 1)
+            # rows, which could be safely dropped.
+            split_indices = [split_index * i for i in range(1, n + 1)]
+            shards = self.split_at_indices(split_indices)
```

Comment:
The final shard may contain more than `split_index` rows, you'll need to truncate with `shards[-1] = shards[-1].limit(split_index)` or something similar.

Reply:
hmm i'm actually creating n + 1 split here and we just dropped the last split; so it's correct here?